WordPress/openverse-frontend
# Fix back to search link for pages in other languages

## Fixes


Fixes #1345 by @AetherUnbound 

## Description


This PR refactors the `VSearchBarLink`, moving path computation to the parent component.

~~It also enables TS-ifying components that use `?inline` SVGs.~~
I tried using 
```
import type { Component } from 'Vue'`
import Chevron from 'path/to/icons/chevron.svg?inline'
...
components: { Chevron: Chevron as Component }
```
It worked locally, but failed in CI (Could not find module Vue or its types), so I just replaced it with a `VIcon`. 
Considering that `nuxtjs/svg` [does not support Nuxt 3](https://github.com/nuxt-community/svg-module/issues/86), it is probably better to gradually replace all the `?inline` svgs with `VIcon` or `<svg>` elements.

## Testing Instructions

Search for something using localized sites (e.g. search-staging.openverse.engineering/ru/search/q=cat) and open one of the results. There should be '_Back to search results_' link at the top, and it should go to the previous page. This should work for audio and images, from _All Content_, _Images_ and _Audio_ search pages.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
